music_assistant/providers/listenbrainz_scrobble/__init__.py

```diff
@@ -0,0 +1,173 @@
+"""Allows scrobbling of tracks with the help of liblistenbrainz."""
+
+# icon.svg from https://github.com/metabrainz/design-system/tree/master/brand/logos
+# released under the Creative Commons Attribution-ShareAlike(BY-SA) 4.0 license.
+# https://creativecommons.org/licenses/by-sa/4.0/
+
+import asyncio
+import time
+from collections.abc import Callable
+from typing import Any
+
+from liblistenbrainz import Listen, ListenBrainz
+from music_assistant_models.config_entries import (
+    ConfigEntry,
+    ConfigValueType,
+    ProviderConfig,
+)
+from music_assistant_models.constants import SECURE_STRING_SUBSTITUTE
+from music_assistant_models.enums import ConfigEntryType, EventType
+from music_assistant_models.errors import SetupFailedError
+from music_assistant_models.event import MassEvent
+from music_assistant_models.playback_progress_report import MediaItemPlaybackProgressReport
+from music_assistant_models.provider import ProviderManifest
+
+from music_assistant.mass import MusicAssistant
+from music_assistant.models import ProviderInstanceType
+from music_assistant.models.plugin import PluginProvider
+
+
+async def setup(
+    mass: MusicAssistant, manifest: ProviderManifest, config: ProviderConfig
+) -> ProviderInstanceType:
+    """Initialize provider(instance) with given configuration."""
+    return ListenBrainzScrobbleProvider(mass, manifest, config)
+
+
+class ListenBrainzScrobbleProvider(PluginProvider):
+    """Plugin provider to support scrobbling of tracks."""
+
+    _client: ListenBrainz = None
+    _currently_playing: str | None = None
+    _on_unload: list[Callable[[], None]] = []
+
+    def _get_config(self) -> dict[str, ConfigValueType]:
+        return {
+            CONF_AUTH_TOKEN: self.config.get_value(CONF_AUTH_TOKEN),
+        }
```

Comment:
Lets not reinvent the wheel as this may cause confusion with MA's config.
I suggest you just feed the dict into the "_get_client" initialization

Reply:
This was inherited from the last.fm scrobbler, not sure why it was done that way there but indeed I see no reason to keep it here.